block/block.go

```diff
@@ -65,7 +66,14 @@ func (block Block) Sign(signer sig.Signer) (SignedBlock, error) {
 }
 
 func (block Block) String() string {
-	return fmt.Sprintf("Block(Header=%s,Round=%d,Height=%d)", base64.StdEncoding.EncodeToString(block.Header[:]), block.Round, block.Height)
+	txHeaders := make([]byte, 32*len(block.Txs))
+	for i, tx := range block.Txs {
+		txHeader := tx.Header()
+		copy(txHeaders[32*i:], txHeader[:])
+	}
+	txHeaderSHA3 := sha3.Sum256(txHeaders)
+	txHeaderB64 := base64.StdEncoding.EncodeToString(txHeaderSHA3[:])
+	return fmt.Sprintf("Block(Header=%s,ParentHeader=%s,Timestamp=%d,Round=%d,Height=%d,TxHeader=%s)", base64.StdEncoding.EncodeToString(block.Header[:]), base64.StdEncoding.EncodeToString(block.ParentHeader[:]), block.Time.Unix(), block.Round, block.Height, txHeaderB64)
```

Comment:
We use `Header=%s` in calculating the header itself. Following our other discussion, we should leave `String()` as is and move this implementation into a separate `func hashBlockIntoHeader(block Block) sig.Hash` function.

Reply:
I've pushed these changes and fixed the tests.